File: frontend/src/data/entities/auction.ts
export type AuctionStatus = 'scheduled' | 'in_progress' | 'completed' | 'cancelled';

export interface Auction {
  id: number;
  venueId: number;
  auctionDate: string;
  startTime?: string;
  endTime?: string;
  status: AuctionStatus;
  createdAt: string;
  updatedAt: string;
}

export interface AuctionItem {
  id: number;
  auctionId: number;
  fishermanId: number;
  fishType: string;
  quantity: number;
  unit: string;
  highestBid?: number;
  highestBidderId?: number;
  highestBidderName?: string;
  sortOrder: number;
  createdAt: string;
  deletedAt?: string;
}

export interface Bid {
  itemId: number;
  buyerId: number;
  price: number;
}

File: backend/internal/server/handler/buyer/response/auction.go
package response

// Auction represents an auction view for the buyer.
type Auction struct {
	ID        int     `json:"id"`
	VenueID   int     `json:"venue_id"`
	StartAt   *string `json:"start_at"`
	EndAt     *string `json:"end_at"`
	Status    string  `json:"status"`
	CreatedAt string  `json:"created_at"`
	UpdatedAt string  `json:"updated_at"`
}
